query-engine/prisma-models/src/field/scalar.rs

```diff
@@ -90,6 +90,8 @@ impl ScalarField {
         };
 
         match scalar_field_type {
+            // TODO: Fix message -- composite types do work?
+            // ? Is this message not validated anywhere ?
             ScalarFieldType::CompositeType(_) => {
                 unreachable!("Cannot convert a composite type to a type identifier. This error is typically caused by mistakenly using a composite type within a composite index.",)
```

Comment:
highlighting

Reply:
The message was added in the bounty PR, before it was just `unreachable!` with no text.
The "unreachable" showed up in the panic when one tried to use that.
(See original issue that the bounty PR tried to fix)

Reply:
No, I know, I just meant a long the lines of: we don't have tests for this message anywhere in engines.
The original issue _only_ has this showing as a result of `generate`

Reply:
With the new validation error, you probably can not run into this any more? Or do I misunderstand?

Reply:
With some testing, I cannot get this error to pop up when _only_ using composite types. I'm certain that this only came up because it was being used in a _`unique()`_ array that contained both a normal scalar _and_ a composite type.

With the validation this is now not a possible case

Reply:
Then it seems like we have two options:

1. Make it empty again:
```suggestion
            ScalarFieldType::CompositeType(_) => {
                unreachable!()
```
2. Put _some_ message in there that is at least not wrong:
```suggestion
            ScalarFieldType::CompositeType(_) => {
                unreachable!("This should not be reached. (Composite types are currently not supported in compound indexes.)",)
```